.github/pull_request_template.md

```diff
@@ -1,15 +1,27 @@
-# Proposed Changes
+## Description (Proposed Changes)
+(Please try to mention in bullet points.)
 
--
 -
 -
 
-Related Issues (if applicable)
+## Link to the issue : 
 
--
+(Add link of the issue you have proposed changes to)
+
+- Link :
+
+## Tests
+
+I added the following tests (mention the file name as well): 
+
+- No Test(s) added.
+
+## Checklist
 
-## Please check that the PR fulfills these requirements
+Please check that the PR fulfills all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.
 
 - [ ] Set a 100 character limit in your editor/IDE to avoid white space diffs in the PR
 - [ ] Tests for the changes have been added (for bug fixes / features)
 - [ ] Added yourself to AUTHORS.md
+- [x] I updated/added relevant documentation (doc comments with `///`).
```

Comment:
Better remove the "x" and the "I" (then all bullet points have the same structure)